### ISSUE TITLE
U CANT CLAMB LADDERS IN OTHER MODS

### ISSUE THREAD

**Issue by iamsoNewBee** [makamys/SmartMoving#5]
YES.THE VANILLA LADDER IS AVAILABLE.BUT OTHER`S CANT.

**Comment by tyrael93** [comment]
Yes, that is an issue with smart moving. Disable it while you try to go up and down a not-vanilla ladder, that works.

**Comment by makamys** [comment]
Can I get some examples of mods that have ladders?

The root of the issue is that Smart Moving has a hardcoded list of surfaces that it lets you climb, rather than letting you climb anything that implements `Block#isLadder` to return true.

I could potentially fix it by making it consider any unknown blocks that claim to be ladders to be ladders too, but the climbing code is kind of a nightmare so we'll see how that goes.

**Comment by tyrael93** [comment]
> The root of the issue is that Smart Moving has a hardcoded list of surfaces that it lets you climb
Ah that would explain it.
I personally only experienced the issue with techguns' ladders that spwan inside the structures, and I just blamed techguns being a bloody mess for that honestly (which it very much is) but I'm guessing 1.7 ender IO does have dark steel ladders? We'll have to wait on the dude to find out what he might be talking about.

**Comment by iamsoNewBee** [comment]
many mods include ladders such as TC or HBM and Extendable-Ladders-Mod.
even u cant climb ropes in RopePlus

**Comment by iamsoNewBee** [comment]
would u like to  just separate the “crawl forward” function？there is no any this kind of mod in 1.7.10

**Comment by makamys** [comment]
No, if I did that I'd want to rewrite the entire mod, and that would be a lot of work for the benefit of achieving slightly better mod compatibility. The best you can do is disable all mod features other than crawling in the config.

**Comment by iamsoNewBee** [comment]
OKAY I KNOW IT

**Comment by DylanLogan2581** [comment]
This also affects the carpenters trapdoors from carpenters blocks. If you have a trapdoor over a ladder you can't get up.

**Comment by ShamelessDuck** [comment]
Maybe, just maybe, a hack for this would be to add a config line to which you can add ladders from other mods so that you won't need to re-compile the mod to add new ladders? Yeah, you still would need to manually add ladders' IDs, not the most user friendly option, but an option nonetheless.

**Comment by Shahelion** [comment]
Would this also cause incompatibility with TreeClimbing? It allows climbing any logs like ladders if there are leaves detected on top.

**Comment by iamsoNewBee** [comment]
> Would this also cause incompatibility with TreeClimbing? It allows climbing any logs like ladders if there are leaves detected on top.

YES
